chalicelib/checks/wfr_checks.py

```diff
@@ -14,22 +14,258 @@
 from .helpers.confchecks import *
 from .helpers.linecount_dicts import *
 
+#-----------------------------------------------------------
+#           Pipelines and Versions supported
+#-----------------------------------------------------------
 default_pipelines_to_run = ['WGS Trio v25', 'WGS Proband-only Cram v25', 'CNV v2']
 
 
+
+#+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+#
+#           !!! FUNCTIONS USED BY CHECKS !!!
+#
+#+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+#+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+#                   General functions
+#+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+############################################################
+# _metadata_for_patch
+############################################################
+def _metadata_for_patch(metawfr_uuid, ff_key):
+    """
+    Fetch meta data for metawfr, sample_processing,
+    sample_processing short (frame=raw)
+    """
+    ### Getting metawfr metadata
+    metawfr_meta = ff_utils.get_metadata(metawfr_uuid, key=ff_key)
+    ### Getting the case that will be patched
+    case_acc = metawfr_meta['title'].split(' ')[-1]
+    case_meta = ff_utils.get_metadata(case_acc, add_on='?frame=raw', key=ff_key)
+    ### Getting the sample_processing that will be modified and used for patching
+    sp_uuid = case_meta['sample_processing']
+    sp_meta = ff_utils.get_metadata(sp_uuid, key=ff_key)
+    sp_meta_short = ff_utils.get_metadata(sp_uuid, add_on='?frame=raw', key=ff_key)
```

Comment:
All of the info in `sp_meta_short` is contained in `sp_meta`, unless in the case of unprocessed invalidation. I would recommend just using `sp_meta` and updating the various code blocks to use that view. If you want to ensure an up to date view of the entire object, use the following.
```
sp_meta_embedded_db = ff_utils.get_metadata(sp_uuid, add_on='frame=embedded&datastore=database', key=ff_key)
```
Note that this operation can be expensive when computed for a very large object. If you only need raw properties and calculated properties (not embeds), replace `embedded` with `object`. If you don't need calculated properties or resolved resource paths, replace `embedded` with `raw`. The differing views require less load on the back-end services, so it is very important to only use `embedded` view especially when absolutely needed. Doug's embed optimization API is a potential option as well. It's very common to require `object` view, but there is no reason I can think of to make calls back-to-back in this manner.

Reply:
I changed the function to use only sp_meta, tried also to merge and simplify two functions and parametrize the pipeline steps as a argument for the functions defined in the action.